packages/analytics-core/src/timeline.ts

```diff
@@ -31,19 +33,24 @@ export class Timeline {
     return Promise.resolve();
   }
 
-  push(event: Event, config: Config) {
+  reset() {
+    this.applying = false;
+    this.isReady = false;
+    this.plugins = [];
+  }
+
+  push(event: Event, config: Config | null) {
     return new Promise<Result>((resolve) => {
-      if (config.optOut) {
-        resolve(buildResult(event, 0, OPT_OUT_MESSAGE));
+      this.queue.push([event, resolve]);
+      if (config === null) {
```

Comment:
do we still need this if we are checking `this.isReady` in `scheduleApply`?

Reply:
Good catch, will remove this.